java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/IPAddress.java

```diff
@@ -169,4 +183,26 @@ else if (address instanceof Inet6Address)
     public static IPAddress<?> fromInetAddress(@Nonnull InetAddress address) {
         return of(address);
     }
+
+    @Nonnull
+    protected <T extends InetAddress> T toInetAddressSubclassChecked(
```

Comment:
I thought about doing it directly in the subclass like this too, which I agree is and looks simpler too.

The only reason I chose to do it this way was to make sure that when such exceptions/errors occur (which almost certainly would have been caused by programming errors), the error message will carry the IP address that it was trying to convert so that we can reproduce for debugging. What do you think?